core/primitives/src/views.rs

```diff
@@ -1653,6 +1653,26 @@ impl ExecutionOutcomeWithIdView {
     }
 }
 
+pub struct TxStatusView {
+    pub execution_outcome: Option<FinalExecutionOutcomeViewEnum>,
+    pub status: TxExecutionStatus,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, serde::Serialize, serde::Deserialize, Clone, Debug)]
+pub enum TxExecutionStatus {
+    /// Transaction is waiting to be included into the block
+    None,
+    /// Transaction is included into the block
+    Inclusion,
+    /// Transaction is included into the block, and the block is finalised
+    InclusionFinal,
+    /// All the receipts (originated by transaction) finished their execution
+    Executed,
```

Comment:
For a field in a response, this enum variant is a bit akward IMO. It says all receipts were executed on some chain but it doesn't even guarantee that the transaction was included in a final block.

Idea: What if the response contains two fields?

```rust
inclusion: No/NotFinal/Final,
execution: No/NotFinal/Final,
```

This approach is not as nice from a strong typing point of view, since it allows encoding invalid states such as `inclusion: No, execution: Final`. But I think it's easier to use than having a single enum that has all valid combinations like this:

```rust
enum TxExecutionStatus {
    None,
    InclusionNonFinal_NotExecuted, // Inclusion
    InclusionNonFinal_ExecutedNonFinal,  // Executed (or Inclusion)
    InclusionFinal_NotExcecuted,  // Inclusion
    InclusionFinal_ExcecutedNonFinal, // Executed (or Inclusion)
    InclusionFinal_ExcecutedFinal, // Final
```

In comments how they map to your current variants.
You can probably find better names than mine to make the meaning clearer. :) But still, I find the two fields approach simpler from an API user perspective.

But yeah, I count 6 valid combinations, where the two-fields approach allow for 9 combinations. Not sure what's the best approach, just writing down all my thoughts :shrug:

Reply:
>  It says all receipts were executed on some chain but it doesn't even guarantee that the transaction was included in a final block.

I am sorry, I wasn't clear enough in these comments, I'll improve it. Current options go from the softest to the strictest. It means, if you received `Executed` response, the requirements for `InclusionFinal` were also satisfied.